packages/api/db/migration/20240115182150_create_animal_breed_tables.js

```diff
@@ -0,0 +1,109 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('default_animal_breed', (table) => {
+    table.increments('id').primary();
+    table.integer('default_type_id').notNullable().references('id').inTable('default_animal_type');
+    table.string('key').notNullable();
+  });
+
+  await knex.schema.createTable('custom_animal_breed', (table) => {
+    table.increments('id').primary();
+    table.uuid('farm_id').notNullable().references('farm_id').inTable('farm');
```

Comment:
You mentioned wanting to check this against the types table custom types?

Reply:
I think when I mentioned that I hadn't realized that custom breeds would need a farm ID because they can belong either to a default or custom type (if they always belonged to default they would just take on the farm ID of the type). Unfortunately I don't think there's any way to enforce this with constraints, the only way would be to add a trigger after inserting the record and deleting it if it doesn't fulfill the criteria.